### DISCUSSION TITLE
Does the nlohmann-json library support incremental parsing of streams?

### DISCUSSION THREAD

**Discussion by RobertHue** [nlohmann/json#2396] [Q&A] [answered]
Hello,

Does the nlohmann-json library support incremental parsing of streams?

Since I read the following inside the appreciation list, it should be possible, or?

> Aaron Burghardt implemented code to parse streams incrementally.

Is there any example out there, that shows it's usage?

**Comment by byronfa** [comment]
I'm stuck asking a similar question it would seem..

**Comment by nlohmann** [comment, marked as answer]
You can parse from a `std::istream` via `operator>>`. It parses until a complete JSON value is read you can call it again to parse the next value. So when your input is `true false`, then you can parse two values.

**Reply by RobertHue** [reply]
Thank you
